### PR TITLE
feat: new option for block mining with local config

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

In order to avoid mining manually during tests and development with a local  node, we should introduce an option for the node start command  that will start miner script if present.

## What was done?

Added a new option `mine-blocks` for the `start` command

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/commands/start.js [hunk 1]
```diff
@@ -22,11 +28,25 @@ class StartCommand extends BaseCommand {
...
     },
     dockerCompose,
     startNodeTask,
     config,
   ) {
+    let blockTimeMs;
+
+    if (mineBlocks !== null) {
+      if (config.get('network') !== NETWORKS.LOCAL) {
+        throw new Error(`mine-blocks option supposed to work only with local network. Your network is ${config.get('network')}`);
```
Comment (shumkov) [inline src/commands/start.js:41]: `this.error` ?

File: src/commands/start.js [hunk 2]
```diff
@@ -22,11 +28,25 @@ class StartCommand extends BaseCommand {
...
+      }
+
+      blockTimeMs = ms(mineBlocks);
+      if (blockTimeMs === undefined || blockTimeMs < 0) {
+        throw new Error(`Invalid mine-blocks value ${mineBlocks}`);
```
Comment (shumkov) [inline src/commands/start.js:46]: `this.error` ?

File: src/commands/start.js [hunk 3]
```diff
@@ -41,6 +61,25 @@ class StartCommand extends BaseCommand {
             },
           ),
         },
+        {
+          title: 'Start mining',
```
Comment (shumkov) [inline src/commands/start.js:65]: Start miner

File: src/commands/start.js [hunk 4]
```diff
@@ -70,6 +109,7 @@ StartCommand.flags = {
   update: flagTypes.boolean({ char: 'u', description: 'download updated services before start', default: false }),
   'drive-image-build-path': flagTypes.string({ description: 'drive\'s docker image build path', default: null }),
   'dapi-image-build-path': flagTypes.string({ description: 'dapi\'s docker image build path', default: null }),
+  'mine-blocks': flagTypes.string({ description: 'new blocks mining interval for local node e.g. 5s', default: null }),
```
Comment (shumkov) [inline src/commands/start.js:112]: blocks mining interval for local node e.g. '5s' or '2.5m'

File: src/commands/start.js [hunk 5]
```diff
@@ -41,6 +61,25 @@ class StartCommand extends BaseCommand {
             },
           ),
         },
+        {
+          title: 'Start mining',
+          enabled: () => mineBlocks !== null,
+          task: async () => {
+            const privateKey = new PrivateKey();
+            const address = privateKey.toAddress('regtest').toString();
```
Comment (shumkov) [inline src/commands/start.js:69]: Let's add an option `mine-blocks-address`? Could be useful.

File: src/commands/start.js [hunk 6]
```diff
@@ -22,11 +28,25 @@ class StartCommand extends BaseCommand {
...
       'dapi-image-build-path': dapiImageBuildPath,
+      'mine-blocks': mineBlocks,
     },
     dockerCompose,
     startNodeTask,
     config,
   ) {
+    let blockTimeMs;
+
+    if (mineBlocks !== null) {
```
Comment (shumkov) [inline src/commands/start.js:39]: Let's add new options to config that could be overwritten by command options.:
 - `core.miner.enabled=false`
 - `core.miner.interval="2.5m"`
 - `core.miner.address=null`

In this case, you don't need to pass them every time.

**Review by shumkov — CHANGES_REQUESTED** [review]

File: src/commands/start.js (hunk 1 above)
Comment (shuplenkov) [reply src/commands/start.js:41]: what does it mean `this.error`?

File: src/commands/start.js (hunk 2 above)
Comment (shuplenkov) [reply src/commands/start.js:46]: what does it mean `this.error`?

File: src/config/configJsonSchema.js [hunk 7]
```diff
@@ -73,6 +73,22 @@ module.exports = {
...
           additionalProperties: false,
         },
+        miner: {
+          type: 'object',
+          properties: {
+            enable: {
+              type: 'boolean',
+            },
+            interval: {
+              type: 'string',
```
Comment (shumkov) [inline src/config/configJsonSchema.js:83]: pattern: ^[0-9]+(\.[0-9]+)?(m|s|h)?$

**Review by shumkov — APPROVED** [review]
Cool! 👍

**Pull request merged**